src/main/scala/com/kjetland/jackson/jsonSchema/JsonSchemaGenerator.scala

```diff
@@ -1045,9 +1082,12 @@ class JsonSchemaGenerator
                 // Push current work in progress since we're about to start working on a new class
                 definitionsHandler.pushWorkInProgress()
 
-                if( (classOf[Option[_]].isAssignableFrom(propertyType.getRawClass) || classOf[Optional[_]].isAssignableFrom(propertyType.getRawClass) ) && propertyType.containedTypeCount() >= 1) {
+                if((classOf[Option[_]].isAssignableFrom(propertyType.getRawClass) ||
+                  classOf[Optional[_]].isAssignableFrom(propertyType.getRawClass) ||
+                  classOf[com.google.common.base.Optional[_]].isAssignableFrom(propertyType.getRawClass))
```

Comment:
I noticed that we weren't covering Guava's Optional type here, so added it in. It seems like the visitor was traversing fine without it so this may not make any functional difference in the generated schema, but makes more sense logically.